examples/pipelines/fl_cross_silo_factory/submit.py

```diff
@@ -0,0 +1,308 @@
+"""Federated Learning Cross-Silo pipeline built by a factory class.
+
+This scripts wraps up the FL pipeline orchestration code in a "factory" class.
+This class is a draft API to build a pipeline based on simple steps
+(preprocessing, training, aggregation). The factory could be extended
+to cover more advanced scenarios.
+
+In sequence, this script will:
+A) reads a config file in yaml specifying the number of silos and their parameters,
+B) reads the components from a given folder,
+C) allow developers to write FL pipeline steps as pythonic functions,
+D) call the factory class to build the full FL pipeline based on custom user code.
+"""
+import os
+import argparse
+import random
+import string
+import datetime
+
+# Azure ML sdk v2 imports
+from azure.identity import DefaultAzureCredential, InteractiveBrowserCredential
+from azure.ai.ml import MLClient, Input, Output
+from azure.ai.ml.constants import AssetTypes
+from azure.ai.ml.dsl import pipeline
+from azure.ai.ml import load_component
+
+# to handle yaml config easily
+from omegaconf import OmegaConf
+
+from typing import List, Optional, Union
+from typing import Callable, Dict
+from dataclasses import dataclass
+import itertools
+
+# local imports
+from fl_factory import FederatedLearningPipelineFactory
+
+###############################
+### A. CONFIGURE THE SCRIPT ###
+###############################
+
+parser = argparse.ArgumentParser(description=__doc__)
+
+parser.add_argument(
+    "--config",
+    type=str,
+    required=False,
+    default=os.path.join(os.path.dirname(__file__), "config.yaml"),
+    help="path to a config yaml file",
+)
+parser.add_argument(
+    "--submit",
+    default=False,
+    action="store_true",
+    help="actually submits the experiment to AzureML",
+)
+
+parser.add_argument(
+    "--example", required=False, choices=["MNIST"], default="MNIST", help="dataset name"
+)
+
+args = parser.parse_args()
+
+# load the config from a local yaml file
+YAML_CONFIG = OmegaConf.load(args.config)
+
+# CONNECT TO AZURE ML
+
+try:
+    credential = DefaultAzureCredential()
+    # Check if given credential can get token successfully.
+    credential.get_token("https://management.azure.com/.default")
+except Exception as ex:
+    # Fall back to InteractiveBrowserCredential in case DefaultAzureCredential not work
+    credential = InteractiveBrowserCredential()
+
+# Get a handle to workspace
```

Comment:
I'm a bit confused about why we try connect to workspace using config.json first, then try config.yaml if that not working.

Suggestion: In my opinion, better to always use the aml workspace in config.yaml for the FL experiment (since that file is the FL experiment config yaml file and thus should be the targeted aml ws). Adding the case of loading ws from config.json file in this submission script might just add confusion/complexity rather than providing clarity/help, and I suggest to remove it from this submission script.

Reply:
This part is common between the factory submit and the native submit. The discussion here could be between 3 ways to provide aml references:
- provide as command line arguments to the submit.py script (ex: `--workspace NAME`)
- provide in yaml
- provide in config.json

Maybe a good order of precedence could be:
- first check cli arguments,
- in no cli, check yaml
- if no yaml, use config

Since this is a common piece of code between native and factory, I propose to file this as an open issue for this september release and figure it out for both scenarios.

Reply:
Filed as #79